underfs/gcs/src/main/java/alluxio/underfs/gcs/GCSUnderFileSystem.java

```diff
@@ -511,7 +538,7 @@ private String getChildName(String child, String parent) {
    */
   private GSObject getObjectDetails(String key) {
     try {
-      if (isFolder(key)) {
+      if (isDirectory(key)) {
```

Comment:
A minor optimization could be to introduce an `isDirectoryInternal` which returns the `GSObject`. Then we could avoid a follow up RPC.